static/app/views/projectDetail/projectDetail.tsx

```diff
@@ -48,158 +49,100 @@ type RouteParams = {
 };
 
 type Props = RouteComponentProps<RouteParams, {}> & {
-  loadingProjects: boolean;
   organization: Organization;
-  projects: Project[];
-  selection: PageFilters;
 };
 
-type State = DeprecatedAsyncView['state'];
-
-class ProjectDetail extends DeprecatedAsyncView<Props, State> {
-  getTitle() {
-    const {params, organization} = this.props;
-
-    return routeTitleGen(t('Project %s', params.projectId), organization.slug, false);
-  }
-
-  componentDidMount() {
-    super.componentDidMount();
-    this.syncProjectWithSlug();
-  }
-
-  componentDidUpdate() {
-    this.syncProjectWithSlug();
-  }
-
-  get project() {
-    const {projects, params} = this.props;
-
-    return projects.find(p => p.slug === params.projectId);
-  }
-
-  handleProjectChange = (selectedProjects: number[]) => {
-    const {projects, router, location, organization} = this.props;
-
-    const newlySelectedProject = projects.find(p => p.id === String(selectedProjects[0]));
-
-    // if we change project in global header, we need to sync the project slug in the URL
-    if (newlySelectedProject?.id) {
-      router.replace(
-        normalizeUrl({
-          pathname: `/organizations/${organization.slug}/projects/${newlySelectedProject.slug}/`,
-          query: {
-            ...location.query,
-            project: newlySelectedProject.id,
-            environment: undefined,
-          },
-        })
-      );
-    }
-  };
-
-  handleSearch = (query: string) => {
-    const {router, location} = this.props;
-    router.replace({
-      pathname: location.pathname,
-      query: {
-        ...location.query,
-        query,
-      },
-    });
-  };
-
-  tagValueLoader = (key: string, search: string) => {
-    const {location, organization} = this.props;
-    const {project: projectId} = location.query;
-
-    return fetchTagValues({
-      api: this.api,
-      orgSlug: organization.slug,
-      tagKey: key,
-      search,
-      projectIds: projectId ? [projectId] : undefined,
-      endpointParams: location.query,
-    });
-  };
-
-  syncProjectWithSlug() {
-    const {router, location} = this.props;
-    const projectId = this.project?.id;
-
+export default function ProjectDetail({router, location, organization}: Props) {
+  const api = useApi();
+  const params = useParams();
+  const {projects, fetching: loadingProjects} = useProjects();
+  const {selection} = usePageFilters();
+  const project = projects.find(p => p.slug === params.projectId);
+  const {query} = location.query;
+  const hasPerformance = organization.features.includes('performance-view');
+  const hasDiscover = organization.features.includes('discover-basic');
+  const hasTransactions = hasPerformance && project?.firstTransactionEvent;
+  const projectId = project?.id;
+  const isProjectStabilized =
+    defined(project?.id) &&
+    project.id === location.query.project &&
+    project.id === String(selection.projects[0]);
+  const visibleCharts = ['chart1'];
```

Comment:
You could memoize `visibleCharts` as it is not just used for iterating but also as a prop for `<ProjectCharts />`.